### ISSUE TITLE
Update relevant code to player integration according to updated player needs

### ISSUE THREAD

**Issue by afranzese99** [uchicago-cs/chiventure#875]
After the completion of issue #835 (meeting with the relevant RPG teams (playerclass, skill trees, and possibly others) and updating the player integration plan documentation) we will need to implement the desired features into the chiventure code. The player struct will likely be updated, in addition to functions that may become outdated with new additions to the player module/wishlist (e.g. `player_new()`in player.h (.c)).

**Comment by WPDolan** [comment]
@afranzese99 and I have completed documentation of the player struct. When implementing the initialization, new, and free functions, we were unsure if `player_free` should also free the associated `class_t` field. After discussion with the playerclass team, we determined the `class_t` struct will not be freed given the playerclass team's potential desire to use classes for multiclassing, descriptions, or other purposes. Given future implementation of the `class_t` struct by the playerclass team, we may need to free this field in the future

**Comment by afranzese99** [comment]
@WPDolan and I have updated functions that take in the `player_t struct` as a parameter (`player_new()`, `player_init()`, `player_free()`). However, the health field in the `player_t` struct is outdated and will be turned into a stat held within the `stats_hash_t *player_stats`, this will break many functions that currently use the health field in the player struct (e.g. `change_health()` in the player module, `chiventures_ctx_init()` in the common module, etc.). We will also need to update wherever `player_new()` is called now that this function takes in different parameters than before.

**Comment by WPDolan** [comment]
After discussing with Professor Borja, I have updated the `player_new` function to only set a the player's ID, instead of every field present in the player struct. I have also updated all function calls to `player_new` from setting both the health and id to only setting the player's health. Currently, the implementation for the health functions remain in the code, but they will most likely be removed once health is fully integrated as a stat in the player's `player_stats` field.

**Comment by WPDolan** [comment]
The pull request has been approved, as the associated branch with this issue, `action-management/player-integration`, has been successfully merged. 

We have removed the old `player_init` function which set the fields of the player struct. In its place, we have created multiple functions to set individual fields in the player struct such as player_set_race and player_set_class. More functions will be added in future issues (#944 and #952) to set other fields in the player struct. This allows developers to ignore fields in the player struct that may not pertain to their game.

We also removed the health field from the old player struct. We intend that both a player's maximum and current health will be stored as respective stats in the player's player_stats hash. This will allow a player's health to be affected by functions currently implemented in the stats module, and provide additional options for its exclusion should a game developer be uninterested in health as a mechanic for their game.
